### ISSUE TITLE
IntuneMobileAppConfigurationPolicyIOS is not needed

### ISSUE THREAD

**Issue by ricmestre** [microsoft/Microsoft365DSC#5459]
### Description of the issue

*IntuneMobileAppConfigurationPolicyIOS* is not needed, this resource is already covered by *IntuneAppConfigurationDevicePolicy* which takes care of both Android and iOS.

### Microsoft 365 DSC Version

1.24.1120.1

### Which workloads are affected

Intune

### The DSC configuration

```powershell
N/A
```

### Verbose logs showing the problem

```powershell
N/A
```

### Environment Information + PowerShell Version

```powershell
Win11/PS5.1
```

**Comment by NikCharlebois** [comment]
@ykuijs  FYI.

**Comment by ykuijs** [comment]
This was new resource was submitted by @dannyKBjj and merged last week in PR #5401.

@dannyKBjj, were you aware of the resource @ricmestre is mentioning and that is was capable of also managing the iOS policies?

If this is a duplicate, we will have to remove the newly added resource again. Working on a PR now and will include this rollback in that PR.

**Comment by dannyKBjj** [comment]
Are you sure? Because I was testing with what I believed to be the latest release and it was not exporting those policies (hence my creating the module),  there are Managed App policies (which might be what this existing module does) and managed device policy which targets the apps (which is what mine does).... I.e. within Intune GUi -  Apps>App Configuration Policies > create > managed devices... Which despite the name brings you 'Create App Configuration Policy' (for managed devices, rather than managed apps)

**Comment by dannyKBjj** [comment]
Unless this other module was in a very recent release (like within last month, as I can't say I've been religiously updating since downloading the module) I don't think it is anything to do with mine.

**Comment by ricmestre** [comment]
@dannyKBjj See below, I'm able to export the same policy *Apple iOS - Outlook Configuration Profile* using both DSC resources, difference is that *IntuneAppConfigurationDevicePolicy* also works for Android policies.

![Image](https://github.com/user-attachments/assets/5173c18f-5be5-4ba0-b406-77b8956085db)

EDIT: And yes both policies are Managed devices policies

**Comment by ricmestre** [comment]
@dannyKBjj *IntuneAppConfigurationDevicePolicy* has been made available back in May in version 1.24.529.1

**Comment by dannyKBjj** [comment]
Fair enough, I have submitted another module to support Android which was also accepted (but is not in latest release). I guess that is also redundant and will need removing (IntuneAndroidManagedStoreAppConfiguration).

**Comment by ricmestre** [comment]
Yep, just looked at your code and *IntuneAndroidManagedStoreAppConfiguration* is also duplicate, I'm not even saying which way is better, but since there's already an older version available for several months that's the one that we should use.

**Comment by dannyKBjj** [comment]
I'm about to submit a load more modules for things I don't think are exporting, I will make sure i'm on latest version and do a re-export of everything from my tenant to double check before I do.

**Comment by ykuijs** [comment]
> Yep, just looked at your code and _IntuneAndroidManagedStoreAppConfiguration_ is also duplicate, I'm not even saying which way is better, but since there's already an older version available for several months that's the one that we should use.

Agree with that. If the older one has issue, we should fix them. If things are missing, these should be added.

Which resource is the _IntuneAndroidManagedStoreAppConfiguration_ overlapping with?

**Comment by dannyKBjj** [comment]
Seems to duplicate the functionality of MSFT_IntuneAppConfigurationDevicePolicy , unsure how or why it was missed as we weren't using a particularly old module version... we created export command line from https://export.microsoft365dsc.com/ by selecting all the Intune options. Have just tried this myself and it includes that module in the list of components so unsure why it wasn't captured. I guess maybe there's some delay between module releases and that tool being updated? Have just created a script to dynamically discover what's available in the DSCResources folder instead.

**Comment by FabienTschanz** [comment]
@ricmestre The resource is now removed, the issue can be closed.

**Comment by ricmestre** [comment]
@ykuijs @FabienTschanz Like I explained *IntuneAppConfigurationDevicePolicy* covers both iOS and Android so now that *IntuneMobileAppConfigurationPolicyIOS* has been removed *IntuneAndroidManagedStoreAppConfiguration* must also be removed.

**Comment by ykuijs** [comment]
Will still have to remove the IntuneAndroidManagedStoreAppConfiguration resource. Will create a PR for that shortly and link this issue, so it will automatically close when the PR is merged
